### PR TITLE
introduced tests for pymc.distributions.logprob._get_scaling_

### LINKED ISSUE #5515: Add test for `_get_scaling`
This function has a somewhat long and complicated sequence of conditional branches. AFAICT there is no direct test, so we should add one that covers all the expected cases to facilitate future maintenance and refactoring.

Bonus points for adding type hints

https://github.com/pymc-devs/pymc/blob/114e4391f6950eb9454381d41cf656b82ce137ea/pymc/distributions/logprob.py#L46-L107

Related to #4582

### PR DESCRIPTION
**Pull request by chritter**
Fixes #5515. We have implemented tests for  pymc.distributions.logprob._get_scaling_ which include edge cases and a test with a model graph random variable. Please let us know if the test coverage is sufficient or if further tests are required.

Co-authored-by: @Icyshaman.

Depending on what your PR does, here are a few things you might want to address in the description:
+ [x] what are the (breaking) changes that this PR makes?
+ [ ] important background, or details about the implementation
+ [ ] are the changes—especially new features—covered by tests and docstrings?
+ [x] [linting/style checks have been run](https://docs.pymc.io/en/latest/contributing/python_style.html)
+ [ ] [consider adding/updating relevant example notebooks](https://github.com/pymc-devs/pymc-examples)
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

#DataUmbrella Sprint
@reshamas

### CODE REVIEW

**Review by michaelosthege — COMMENTED** [review]
Looking quite good.
I'm not familiar with `_get_scaling` so I can't tell if the numbers are correct, but I assume that they are.

Can you add `match="part of the message"` to the `pytest.raises(...)`? That'll make sure that it's testing the right errors.

File: pymc/tests/test_logprob.py [hunk 1]
```diff
@@ -41,6 +41,44 @@ def assert_no_rvs(var):
...
+def test_get_scaling():
+
+    assert _get_scaling(None, (2, 3), 2).eval() == 1
+    # ndim >=1 & ndim<1
+    assert _get_scaling(45, (2, 3), 1).eval() == 22.5
+    assert _get_scaling(45, (2, 3), 0).eval() == 45
+
+    # list or tuple tests
+    # total_size contains other than Ellipsis, None and Int
+    with pytest.raises(TypeError):
```
Comment (michaelosthege) [inline pymc/tests/test_logprob.py]:
```suggestion
    with pytest.raises(TypeError, match="expected int or list of ints"):
```

File: pymc/tests/test_logprob.py [hunk 2]
```diff
@@ -41,6 +41,44 @@ def assert_no_rvs(var):
...
+        _get_scaling([2, 4, 5, 9, 11.5], (2, 3), 2)
+    # check with Ellipsis
+    with pytest.raises(ValueError):
+        _get_scaling([1, 2, 5, Ellipsis, Ellipsis], (2, 3), 2)
+    with pytest.raises(ValueError):
+        _get_scaling([1, 2, 5, Ellipsis], (2, 3), 2)
```
Comment (michaelosthege) [inline pymc/tests/test_logprob.py]:
```suggestion
    with pytest.raises(ValueError, match="Double Ellipsis"):
        _get_scaling([1, 2, 5, Ellipsis, Ellipsis], (2, 3), 2)
    with pytest.raises(ValueError, match="scalings is bigger that ndim"):
        _get_scaling([1, 2, 5, Ellipsis], (2, 3), 2)
```

File: pymc/tests/test_logprob.py [hunk 3]
```diff
@@ -41,6 +41,44 @@ def assert_no_rvs(var):
...
+
+    assert _get_scaling([Ellipsis], (2, 3), 2).eval() == 1
+
+    assert _get_scaling([4, 5, 9, Ellipsis, 32, 12], (2, 3, 2), 5).eval() == 960
+    assert _get_scaling([4, 5, 9, Ellipsis], (2, 3, 2), 5).eval() == 15
+    # total_size with no Ellipsis (end = [ ])
+    with pytest.raises(ValueError):
```
Comment (michaelosthege) [inline pymc/tests/test_logprob.py]:
```suggestion
    with pytest.raises(ValueError, match="scalings is bigger that ndim"):
```

File: pymc/tests/test_logprob.py [hunk 4]
```diff
@@ -41,6 +41,44 @@ def assert_no_rvs(var):
...
+        _get_scaling([1, 2, 5], (2, 3), 2)
+
+    assert _get_scaling([], (2, 3), 2).eval() == 1
+    assert _get_scaling((), (2, 3), 2).eval() == 1
+    # total_size invalid type
+    with pytest.raises(TypeError):
```
Comment (michaelosthege) [inline pymc/tests/test_logprob.py]:
```suggestion
    with pytest.raises(TypeError, match="expected int or list of ints"):
```

**Comment by michaelosthege** [comment]
@ricardoV94 @chritter can you check/fix/apply the suggestions I commented? Let's get this over the finish line

**Review by michaelosthege — APPROVED** [review]

**Review by michaelosthege — APPROVED** [review]
I rebased it to fix git conflicts.

Will merge once the tests pass.
